externs/es3.js

```diff
@@ -1906,8 +1906,8 @@ String.prototype.length;
 
 /**
  * @constructor
- * @param {*=} opt_pattern
- * @param {*=} opt_flags
+ * @param {(RegExp|string)=} opt_pattern
+ * @param {string=} opt_flags
```

Comment:
these changes don't work, unfortunately. The type of RegExp is hard-coded in the compiler, and since some people use the head externs with the released compiler we can't easily update both at once. There is a way to do this (I don't remember what it is), but for now I'll drop this and apply the rest of the change.